packages/tempus/unit_test/Tempus_UnitTest_TimeEvent.cpp

```diff
@@ -0,0 +1,1080 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#include "Teuchos_UnitTestHarness.hpp"
+#include "Teuchos_XMLParameterListHelpers.hpp"
+#include "Teuchos_TimeMonitor.hpp"
+#include "Teuchos_DefaultComm.hpp"
+
+#include "Tempus_TimeEventBase.hpp"
+#include "Tempus_TimeEventComposite.hpp"
+#include "Tempus_TimeEventRange.hpp"
+#include "Tempus_TimeEventRangeIndex.hpp"
+#include "Tempus_TimeEventList.hpp"
+#include "Tempus_TimeEventListIndex.hpp"
+
+#include "../TestUtils/Tempus_ConvergenceTestUtils.hpp"
+
+#include <cmath>
+
+static double PI = M_PI;
+
+namespace Tempus_Unit_Test {
+
+using Teuchos::RCP;
+using Teuchos::rcp;
+using Teuchos::rcp_const_cast;
+using Teuchos::rcp_dynamic_cast;
+
+
+// ************************************************************
+// ************************************************************
```

Comment:
can we start adding a more descriptive comment on what is being tested?

Reply:
Yeah, I agree.  I just got tired of making unit tests.  I will go through break up these tests into smaller focused tests with a more descriptive comment/name.